### ISSUE TITLE
CF Origin DNS errors when using ALIAS with subdomains

### ISSUE THREAD

**Issue by jhmckimm** [StackExchange/dnscontrol#2892]
**Describe the bug**
When using `ALIAS` with the Cloudflare provider, you can specify a defined subdomain and the configuration will apply, but this is invalid and results in origin DNS errors.

Example code:
```
D("my.domain", REG_NAMECHEAP, DnsProvider(DSP_CF), CF_PROXY_DEFAULT_ON,
    A("web-public", "127.0.0.1"),
    ALIAS("@", "web-public")
);
```

This will apply (well, it would if I wasn't using `127.0.0.1`, CF prevents that), thus causing the @ record to point to an invalid domain (just `web-public` from Cloudflare's perspective.

**Expected behavior**
Such calls would result in an error that forces the use of a FQDN with trailing period.

**DNS Provider**
 - Cloudflare

**Comment by tlimoncelli** [comment]
CC @tresni (maintainer of the cloudflare provider)

Hi Friend!

Thanks for reporting this!

I'm not sure I completely understand the issue.  I think what I hear you saying is:

```
ALIAS("@", "web-public")
```

...should be the same as:

```
ALIAS("@", "web-public.my.domain.")
```

however instead it is acting like:

```
ALIAS("@", "web-public.")
```

Is that the issue?

**Comment by jhmckimm** [comment]
Hi @tlimoncelli, thanks for getting back to me. Yes - that is indeed my issue.

Since I wrote this, I feel like I have imagined a time where that worked as I expected (with other providers), but looking over some of my DnsControl repos, I can't find any examples - so maybe I am just misusing the feature.

Still, it does seem a bit unintuitive. It would be nice if DnsControl could just infer that it's a subdomain, OR if it could just force ending with a `.`  (and this requiring a FQDN) so others aren't caught out by this.

I suspect this might be down to Cloudflare being a bit of a unique case with `ALIAS`, but I don't know enough about the internals of DnsControl to know for sure.

Cheers!

**Comment by tlimoncelli** [comment]
I consider this a bug.  The target of ALIAS should be treated the same way as the targets of CNAME/MX/NS and so on.

The bad news is that I'm not sure if this is a bug with Cloudflare or DNSControl in general.

The good news is that we have a workaround: Just use a FQDN+"." in the target.

**Comment by jhmckimm** [comment]
Cloudflare just seems to treat whatever you give it as a FQDN. The following configuration:

```js
ALIAS("www2", "web-public.example.org."),
ALIAS("www3", "web-public"),
```

produces:
![image](https://github.com/StackExchange/dnscontrol/assets/6583812/b0936d00-ac04-4f97-ae1e-83dd7b45735b)

The result is that www2 resolves correctly, but www3 gives us a lovely `Error 1016 - Origin DNS error` because `web-public` doesn't resolve.

I suppose this would _actually be a valid use case_ in the case of using a local DNS server (with BIND provider) that has records that exist purely as TLDs (e.g. `https://pihole`), so forcing a terminating `.` is likely the best approach - but, again, I'm not the most experienced with this stuff so I'm likely not taking everything into account.

If `ALIAS` can be adapted just for Cloudflare without affecting other providers, then the expected behaviour could be implemented without issue, because expecting Cloudflare to resolve addresses in the example above would never work.

A bit of a rant but I hope all that makes sense?

**Comment by tlimoncelli** [comment]
Yup!  I think that fixed it.  It was a global problem, not just an issue for Cloudflare.

Could you test it out?

```
git clone https://github.com/StackExchange/dnscontrol.git
cd dnscontrol
git checkout tlim_issue2892_alias
go install
```

This will put a version of dnscontrol in your `~/bin` directory.  Try that with your dnsconfig.js and see if it resolves the problem.

Thanks!
Tom
